<div style="width:100%; height:3em">  </div>

 
<h2><span style="color:#A8A8A8;">   Actividad #01:</span></h2> 
 <strong><br><br><span style="font-size: 4em; margin-left:2em;text-align:center">El Efecto Fotoeléctrico</span><br><br></strong> 
 <br><br><br>
<ul style="list-style-type:none;   float:left">
        <li>   Asignatura:  <strong>Fundamentos Físicos de la Informática </strong></li>
        <li>     Profesor:  <strong>  Alberto Corbi Bellot</strong> </li>
        <li>   Fecha:   <strong>02 de diciembre de de 2020 </strong> </li> 
        <li>    Autores: 
               <ul style="list-style-type:none;  padding-left:5em">
                        <li> <strong> Anyel Martich </strong></li>
                        <li> <strong> Alex Sun </strong></li>
                        <li> <strong>Joffe Pilco</strong> </li>
                        <li>  <strong>Jaime Orozco </strong></li>
            </ul></li></ul>
       
            

   <br><br><br><br><br>   
### Índice
 <ul style="list-style-type:none; margin-left:3%; margin-top:2em">    
     <li><a style="text-decoration: none;" href="#calentamiento">1.- Ejercicio de calentamiento</a></li>    
     <li style="margin-top:1em;"><a  style="text-decoration: none;" href="#ejercicio1">2.- Primer Ejercicio: Laboratorio virtual</a></li>         
    <ul style="list-style-type:none;">
        <li  style="margin-top:1em"><a style="text-decoration: none;" href="#sheep">2.1.- ¿Qué hace CheepJ?</a></li>
        <li  style="margin-top:1em"><a style="text-decoration: none;" href="#relacionmm">2.2.- Relación matemática entre longitud de onda y frecuencia de una onda electromagnética</a></li>
        <li  style="margin-top:1em"><a style="text-decoration: none;" href="#cheepweb">2.2- ¿En qué estándares Web se basa?</a></li>
         <li  style="margin-top:1em"><a style="text-decoration: none;" href="#tabla">2.3- Tabla: [Color-Longitud de onda - frecuencia]</a></li>
      <li  style="margin-top:1em"><a style="text-decoration: none;" href="#ecuacion">2.4- Comprobación Ecuación</a></li>
      
      <li  style="margin-top:1em"><a style="text-decoration: none;" href="#unpkg">2.5- UNPKG / CDN</a></li>
      
      
     </ul>
    
     <li  style="margin-top:1em"><a style="text-decoration: none;" href="#ejercicio2">3.- Segundo Ejercicio: Ajuste a una recta </a></li>
      <li  style="margin-top:1em"><a style="text-decoration: none;" href="#ejercicio3">4.- Tercer Ejercicio: Representación gráfica de los datos</a></li>
      <li  style="margin-top:1em"><a style="text-decoration: none;"  href="#premium">5.- Ejercicio "Premium": Intensidad de corriente en función de la intensidad de la luz</a></li>
    <li  style="margin-top:1em"><a style="text-decoration: none;" href="#historia">6.- Historia en torno al efecto fotoeléctrico</a></li>
     <li  style="margin-top:1em"><a style="text-decoration: none;" href="#valoraciongrupo">7.- Tabla de valoración individual</a></li>
     <li  style="margin-top:1em"><a style="text-decoration: none;" href="#bibliografia">8.- Bibliografía</a></li>
     
  </ul>
   <br><br><br><br><br>

### Comprobad que $e*V$ tiene unidades de energía, donde $e$ es la carga del electrón.


Si conocemos la Fuerza Eléctrica su Fórmula es :


$$\hspace{5cm} F=\frac{K*Q^2}{R^2}(N)$$

Y también la Fórmula de energía de Potencial Eléctrico :
$$\hspace{5cm} Ep=\frac{K*Q^2}{R}(J)$$

Operamos con el método de sustitución en ambas ecuaciones obtenemos:

$$\hspace{5cm}F=\frac{Ep}{R}(N)$$

Conocemos la Fórmula del Incremento de Potencial Eléctrico :

$$\hspace{5cm}F=\frac{V*Q}{R}(N)$$

Operamos con el método de igualación con la fuerza y obtenemos :

$$\hspace{5cm}Ep(J)=V(\frac{N*m}{C})*Q(C)$$

Con ello vemos que ($V*Q$) tiene unidades de Energía.

##    <a id="calentamiento"></a>Ejercicio de calentamiento

#### Demostrar la siguiente afirmación:


$$\hspace{5cm}\frac{x - x_1}{x_2 - x_1} = \frac{y - y_1}{y_2 - y_1 } \equiv y= m \cdot x + b $$

Un <strong>ajuste a una recta</strong> (o regresión lineal) lo que persigue es hallar la ecuación de la recta que mejor pasa por una serie de puntos. Recordamos que una recta se puede definir, como mínimo, con <strong>dos puntos</strong> $x_1,y_1$ y $x_2,y_2$, de manera que cualquier otro punto de dicha recta $x,y$ ,  cumple que: 


$$\hspace{5cm}\frac{x - x_1}{x_2 - x_1} = \frac{y - y_1}{y_2 - y_1 }$$

Esta igualdad también se puede expresar como: $ y= m \cdot x + b $ donde $ m=\frac{y_1-y_2}{x_1-x_2}\  \hspace{.2cm}$y$\hspace{.2cm}   b=\frac{x_1y_2-x_2y_1}{x_1-x_2}$

#### Demostración: 
<br>
<div style="float:left"> 
Despejando:

$$\hspace{0,6cm}\frac{x - x_1}{x_2 - x_1} = \frac{y - y_1}{y_2 - y_1 }; \hspace{0.6cm}y-y_1=\frac{(x - x_1)(y_2 - y_1 )}{x_2 - x_1}; \hspace{0.6cm}y-y_1=\frac{xy_2 -xy_1-x_1y_2+x_1y_1}{x_2 - x_1};$$
Pasamos término $ y_1 $ a la fracción:

$$\hspace{0,6cm}y=\frac{xy_2 -xy_1-x_1y_2+x_1y_1}{x_2 - x_1} + y_1; \hspace{0.6cm}y=\frac{xy_2 -xy_1-x_1y_2+x_1y_1+y_1({x_2 - x_1})}{x_2 - x_1};$$
Operamos y cancelamos el término $ x_1y_1 $ con $ -y_1x_1 $ :

$$\hspace{0.6cm}y=\frac{xy_2 -xy_1-x_1y_2+ {x_1y_1}+y_1x_2 - {y_1x_1}}{x_2 - x_1}; \hspace{0.6cm}y=\frac{xy_2 -xy_1-x_1y_2+y_1x_2}{x_2 - x_1};$$
Factor común a $ x $ y multiplicamos arriba y abajo de los quebrados por $ -1 $:

$$\hspace{0.6cm}y=\frac{y_2 -y_1}{x_2 - x_1} x -\frac{x_1y_2+y_1x_2}{x_2 - x_1}; \hspace{0.6cm}y=\frac{-1\cdot (y_2 -y_1)}{-1\cdot (x_2 - x_1)} x -\frac{-1\cdot(x_1y_2+y_1x_2)}{-1\cdot(x_2 - x_1)};$$
Resultado:

$$\hspace{3.6cm}y= \underbrace{\frac{y_1 -y_2}{x_1 - x_2}}_{m}\cdot x + \underbrace{\frac{x_1y_2 -x_2y_1}{x_1- x_2}}_{b} $$

<a id='ejercicio1'></a>
<h2><span style="font-size:1.5em;"><u>Laboratorio virtual:</u></span><h2>


###  <a id="sheep"></a>¿Qué hace CheepJ tras el telón y en qué estándares Web se basa?

Lo que hace CheerpJ tras el telón es convertir cualquier aplicación en HTML5, WebAssembly, JavaScript.
El funcionamiento es desde el código de bytes de Java (compatible 100%).

CheerpJ reparte muchas funciones de sistema sobre un entorno de ejecución OpenJDK completo y sin cambiar: es accesible para Swing, Filesystem, Audio, Printing y puede conseguir un magnífico rendimiento en tiempo de ejecución y descarga.

# Toma de medidas con el applet del efecto Fotoeléctrico

## Primer ejercicio. 

Haced una tabla rápida con el nombre de cada color del arcoíris (azul, rojo, amarillo, etc.) en una columna, en una segunda columna escribe su longitud de onda (en metros) y en una tercera columna, su frecuencia (en Hercios). 

### <a id="relacionmm"></a> ¿Cuál es la relación matemática entre longitud de onda y frecuencia de una onda electromagnética?

La relación entre longitud de onda y frecuencia es gracias a la velocidad. Sabemos que la velocidad esta expresada en ($\frac{m}{s}$) , la longitud de onda está representada en $m$ y la frecuencia en $Hz$ que es igual al ($\frac{1}{s}$). Es decir, que la expresión es :

$$\hspace{5cm} {velocidad(v)=longitud\hspace{0.1cm} de\hspace{0.1cm}onda(\lambda ) * frecuencia(f)}=m/s$$


la velocidad es una magnitud física que expresa la relación entre el espacio recorrido en un tiempo. En este caso el espacio recorrido es la longitud de onda que es la distancia que recorre un ciclo. Y la frecuencia de onda es el numero de veces que se repite una onda en ese ciclo.

Con la fórmula podemos ver a longitud de onda constante y mayor frecuencia aumenta la velocidad, a menor frecuencia disminuye la velocidad y cuando la frecuencia es constante a mayor longitud de onda mayor velocidad, y a menor longitud de onda menor velocidad.


<a id='ecuacion'></a>

### La ecuación anterior se puede reescribir como (compruébalo tú mismo):

lo comprobamos de la siguiente manera. La fórmula de la velocidad es:

$$\hspace{5cm}v=\frac{d(m)}{t(s)}$$


Al saber las unidades de cada uno:

$$\hspace{5cm}F=Hz=\frac{1}{s}$$

$$\hspace{5cm}v=\frac{m}{s}$$

$$\hspace{5cm}\lambda=m$$


Sustituimos las unidades en la formula :

$$\hspace{5cm}v=F*\lambda$$



<a id='unpkg'></a>

### ¿Para qué sirve UNPKG y por qué es uno de los servicios más usados de Internet (junto con jspm, ESM, cdnjs, jsdelivr, …)? ¿Qué son los CDN?

In [ ]:
UNPKG es una red que envía y recibe información mundial en forma de URL y se envía a cualquier archivo de cualquier paquete de manera eficaz y rápida para todo en NPM.

Porque todos los servicios mas usados de internet tienen a su disposición un servicio o módulo de CDN y que permiten cargar cualquier paquete NPM lo mas rápido posible.

CDN (Content Delivery Netmork) es una red de entrega por contenidos conjuntos de distintos dispositivos. Es decir, busca información de servidores ubicados en diferentes sitios que tienen información de otros servidores para que le lleguen al usuario aquellos datos que quiera los datos mas cercanos a su ubicación.

#### Experimentando con Sodio
  $ \lambda = $ Longitud de onda | $ V = $ Potencial de Frenado 
 :------: |:----------: 
 $331{nm}$| $ -1.40_V $  
         |          |             |             |
 $368{nm}$| $  -1.00_V $   
         |          |             |             |
 $415{nm}$   |$  -0.60_V $ 
         |          |             |             | 
$466{nm}$ |$  -0.20_V $ 
  

  $$Sodio \hspace{1em} \lambda = 331{nm}; V = -1.40_V$$ | $$ Sodio \hspace{1em} \lambda = 368{nm};  V = -1.00_V$$  
  :----------: |:----------:  
<img src="https://github.com/JaimeOro/FFI-Actividad01/blob/main/sodio-331-140.gif?raw=true" style="float:left; height: 320px; margin-right: 1% ; margin-bottom: 0.0em"> | <img src="https://github.com/JaimeOro/FFI-Actividad01/blob/main/sodio-368-100.gif?raw=true" style="float:right; height: 320px; margin-right: 1%; margin-top: 0.0em;">   

  $$Sodio \hspace{1em} \lambda = 415{nm}; V = -0.60_V$$ | $$ Zinc \hspace{1em} \lambda = 466{nm};  V = -0.20_V$$  
  :------: |:----------:  
<img src="https://github.com/JaimeOro/FFI-Actividad01/blob/main/sodio-415-060.gif?raw=true" style="float:left; height: 320px; margin-right: 1% ; margin-bottom: 0.0em"> | <img src="https://github.com/JaimeOro/FFI-Actividad01/blob/main/sodio-466-020.gif?raw=true" style="float:right; height: 320px; margin-right: 1%; margin-top: 0.0em;">   

In [1]:
var longitudes_de_onda_medidas_sodio = [331, 368, 415, 466].map(x => x*1e-9);  // para pasar a metros
var voltajes_medidos_sodio = [-1.40, -1.00, -0.60, -0.20];  // en voltios (voltaje de frenado)

console.log('Logitudes de ondas medidas Sodio= ' + longitudes_de_onda_medidas_sodio);

var inversa_de_longitudes_de_onda_sodio = longitudes_de_onda_medidas_sodio.map(lambda => 1/lambda );  
var datos_a_ajustar_sodio = voltajes_medidos_sodio.reduce((tuplas_longitud_onda_voltaje_sodio, b, i) => {
    tuplas_longitud_onda_voltaje_sodio.push([inversa_de_longitudes_de_onda_sodio[i], b]);
  return tuplas_longitud_onda_voltaje_sodio;
}, []);
console.log('Tuplas 1/lambdas= ' + datos_a_ajustar_sodio); // Verás que son tuplas de valores 1/lambda y voltajes

Logitudes de ondas medidas Sodio= 3.3100000000000004e-7,3.68e-7,4.1500000000000005e-7,4.66e-7
Tuplas 1/lambdas= 3021148.036253776,-1.4,2717391.304347826,-1,2409638.5542168673,-0.6,2145922.746781116,-0.2


#### Experimentando con  Zinc
  $ \lambda = $ Longitud de onda | $ V = $ Potencial de Frenado 
 :------: |:----------: 
 $215{nm}$| $ -1.40_V $  
         |          |             |             |
 $230{nm}$| $  -1.00_V $   
         |          |             |             |
 $246{nm}$   |$  -0.60_V $ 
         |          |             |             | 
$262{nm}$ |$  -0.20_V $ 

  $$Zinc \hspace{1em} \lambda = 215{nm}; V = -1.40_V$$ | $$ Zinc \hspace{1em} \lambda = 230{nm};  V = -1.00_V$$  
  :----------: |:----------:  
<img src="https://github.com/JaimeOro/FFI-Actividad01/blob/main/zinc-215-140.gif?raw=true" style="float:left; height: 320px; margin-right: 1% ; margin-bottom: 0.0em"> | <img src="https://github.com/JaimeOro/FFI-Actividad01/blob/main/zinc-230-100.gif?raw=true" style="float:right; height: 320px; margin-right: 1%; margin-top: 0.0em;"> 

  $$Zinc \hspace{1em} \lambda = 246{nm}; V = -0.60_V$$ | $$ Zinc \hspace{1em} \lambda = 262{nm};  V = -0.20_V$$  
  :------: |:----------:  
<img src="https://github.com/JaimeOro/FFI-Actividad01/blob/main/zinc-246-060.gif?raw=true" style="float:left; height: 320px; margin-right: 1% ; margin-bottom: 0.0em"> | <img src="https://github.com/JaimeOro/FFI-Actividad01/blob/main/zinc-262-020.gif?raw=true" style="float:right; height: 320px; margin-right: 1%; margin-top: 0.0em;">

In [2]:
var longitudes_de_onda_medidas_zinc = [215, 230, 246, 262].map(x => x*1e-9);  // para pasar a metros
var voltajes_medidos_zinc = [-1.40, -1.00, -0.60, -0.20];  // en voltios (voltaje de frenado)


console.log('Logitudes de ondas medidas Zinc= ' + longitudes_de_onda_medidas_zinc);

var inversa_de_longitudes_de_onda_zinc = longitudes_de_onda_medidas_zinc.map(lambda => 1/lambda );  
var datos_a_ajustar_zinc = voltajes_medidos_zinc.reduce((tuplas_longitud_onda_voltaje_zinc, b, i) => {
    tuplas_longitud_onda_voltaje_zinc.push([inversa_de_longitudes_de_onda_zinc[i], b]);
  return tuplas_longitud_onda_voltaje_zinc;
}, []);
console.log('Tuplas 1/lambdas= ' + datos_a_ajustar_zinc); // Verás que son tuplas de valores 1/lambda y voltajes




Logitudes de ondas medidas Zinc= 2.15e-7,2.3000000000000002e-7,2.46e-7,2.62e-7
Tuplas 1/lambdas= 4651162.790697674,-1.4,4347826.0869565215,-1,4065040.650406504,-0.6,3816793.893129771,-0.2


## <a id='ejercicio2'></a> Ejercicio 2: Ajuste a una recta 

### Realizamo el ajuste de la recta y calculamos los valores de $ m $ y $b$

**Constantes:**<br>
$\pi=3.141592 $<br>
$ e=-1.602  x  10^{-19}$<br>
$ c=3 x 10^{8}$<br>

In [3]:
pi=3.141592
e=-1.602e-19
c=3e8

300000000

### Regresión Sodio

In [4]:
var regression = require('./regression.js');
var result = regression.linear(datos_a_ajustar_sodio, { precision: 15 });
var m = result.equation[0];  // pendiente
var b = result.equation[1];  // ordenada al origen
console.log('Pendiente m = ' + m);
console.log('Ordenada b = ' + b);

//var h=(m * 1.602e-19)/(2 * 3.141592 * 3e8);
var h=(m * e)/(2*pi * c)
console.log('h = ' + h + " j*s");

Pendiente m = -0.000001361958109
Ordenada b = 2.705033460922165
h = 1.157511271683274e-34 j*s


Una  vez calculado los datos obtenemos los valores de la pendinente mediante la formula:
$$\hspace{5.3cm} m=\frac{2\pi h c }{ e}$$ y si despejamos de la formula $h$ obtenemos el valor de la consntante de Planck:

$$ h = \frac{m*e}{2*\pi*c};   \hspace{1.5em}   h=\frac{ -0.000001361958109*1.602*10^-19}{2*3.141592*3*10^8}= -1.157511271683274*10^-34\  J*s $$</pre>

### Regresión Zinc

In [5]:
var regression = require('./regression.js');
var result = regression.linear(datos_a_ajustar_zinc, { precision: 15 });
var m = result.equation[0];  // pendiente
var b = result.equation[1];  // ordenada al origen
console.log('Pendiente m = ' + m);
console.log('Ordenada b = ' + b);
//var h=(m * 1.602e-19)/(2 * 3.141592 * 3e8)
var h=(m * e)/(2*pi * c)
console.log('h = ' + h + " j*s");

Pendiente m = -0.000001432967943
Ordenada b = 5.247419703502382
h = 1.2178616471553278e-34 j*s


Una  vez calculado los datos obtenemos los valores de la pendinente mediante la formula:
$$\hspace{5.3cm} m=\frac{2\pi h c }{ e}$$ y si despejamos de la formula $h$ obtenemos el valor de la consntante de Planck:

$$ h = \frac{m*e}{2*\pi*c};   \hspace{1.5em}  h=\frac{ -0.000001432967943*1.602*10^-19}{2*3.141592*3*10^8}= -1.2178616471553278*10^-34\  J*s $$

# Representación gráfica de los datos


In [6]:
function Plotly(data, layout, $$) {
    $$ = $$ || global.$$;
    $$.mime({ "application/vnd.plotly.v1+json": { data: data,  layout: layout } });
}

Para dibujar la recta, tenemos que calcular los puntos $y$ (voltajes) para los mismos valores de $x$ (inversas de las longitudes de onda medidas con el applet) y con los resultados de $m$ y $b$ que hemos obtenido con el ajuste anterior. Matemáticamente: $V_{calculando} = m \cdot \frac 1{\lambda} + b$. 


En Javascript este cálculo se puede hacer de manera muy sencilla mediante el método $map$ sobre el array de frecuencias medidas, como se ve en el codigo siguiente.

In [7]:
voltajes_de_frenado_calculados_sodio = inversa_de_longitudes_de_onda_sodio.map(inversa_lambda => inversa_lambda * m + b);


[
  0.9182114164933202,
  1.3534850757849912,
  1.7944849010927442,
  2.172381199210537
]

In [8]:
voltajes_de_frenado_calculados_zinc = inversa_de_longitudes_de_onda_zinc.map(inversa_lambda => inversa_lambda * m + b);

[
  -1.417547473241803,
  -0.982875700845443,
  -0.5776532355220079,
  -0.22192359039074727
]

Como vemos a continuación, los datos experimentales para el Sodio, se aprecia una gráfica similar, a la de los datos caluculados con la fórmula  $1/\lambda$. Sin ambargo, al aplicarse la inversa la gráfica está en el lado negativo del plano cartesiano.

In [9]:
datos = [
  { x: inversa_de_longitudes_de_onda_sodio, y: voltajes_medidos_sodio, name: 'Datos experimentales' }, 
  { x: inversa_de_longitudes_de_onda_sodio, y: voltajes_de_frenado_calculados_sodio, name: 'Datos calculados tras ajuste' },
];

var layout = {
  title: 'Regresión Sodio<br>x=1/lambda*1/m ',
  xaxis: { title: 'x = Inversa de longitudes',
    titlefont: { family: 'Arial, monospace', size: 15, color: '#c94c4c' , type: 'scatter', autorange: true, uirevision:'true' }
  },
  yaxis: {
    title: 'y= Potencia en Voltage', titlefont: {
      family: 'Arial, monospace ', size: 15, color: '#c94c4c' , type: 'scatter', autorange: true, uirevision:'true' }
  }
};
Plotly(datos, layout);

Para el Zinc no hay mucha diferencia entre los datos experimentales y los datos caluculados con la fórmula $1/\lambda$, ya que al aplicarse la inversa a valores igual a 0 no causa una pendiente diferente a si tuvieramos invirtiendo valores > 0.

In [10]:
datos = [
  { x: inversa_de_longitudes_de_onda_zinc, y: voltajes_medidos_zinc, name: 'Datos experimentales' }, 
  { x: inversa_de_longitudes_de_onda_zinc, y: voltajes_de_frenado_calculados_zinc, name: 'Datos calculados tras ajuste' },
];


var layout = {
  title: 'Regresión Zinc<br>x=1/lambda*1/m ',
  xaxis: { title: 'x = Inversa de longitudes',
    titlefont: { family: 'Arial, monospace', size: 15, color: '#c94c4c' , type: 'scatter', autorange: true, uirevision:'true' }
  },
  yaxis: {
    title: 'y= Potencia en Voltage', titlefont: {
      family: 'Arial, monospace ', size: 15, color: '#c94c4c' , type: 'scatter', autorange: true, uirevision:'true' }
  }
};
Plotly(datos, layout);

###  <a id="premium"></a>Intensidad de corriente en función de la intensidad de la luz

Realizamos el experimento:




<img src="https://raw.githubusercontent.com/amartich/FFI_UNIR/main/Efecto_Fotoelectrico/premium2.gif" style="width: 65%; margin-left: 16%; margin-top: -1.5em;">

Comprobamos en este experimento que al aumentar la intensidad de la luz, aumenta la intensidad de la corriente fotoelectrica.<br>
Observamos además en los datos que ésta relación es proporcional. 

Metal: Zinc<br>
$ \lambda = $ 100nm

 | Intensidad de la luz<br>en banco de pruebas | Intensidad de corriente | Observaciones
 :------: |:----------: | :---------- |
       0%| $ 0.00A $ |Si la intensidad de la luz es 0, no hay emisión de fotones|
         |          |             |             |
      50%| $  0.441A $ | A un 50% de intensidad la corriente es de 0.441A 
      100%| $  0.882A $ | A un 100% de intensidad de la luz, vemos que la corriente es proporcional a la intensidad de la luz.<br> $ \frac{100}{50} = \frac{0.882}{0.441} = 2  $



##    <a id="historia"></a>Historia en torno al Efecto Fotoeléctrico


<span style="float: left; width:100%;"> 
  <br>
     <p>En 1864, James Clerk Maxwell consiguió la unificación de la electricidad y el magnetismo a través de un conjunto de ecuaciones matemáticas que describían el comportamiento del fenómeno que denominó electromagnetismo. En 1887, H. Hertz experimentó la existencia de las ondas electromagnéticas, pero no fue capaz de esbozar ninguna teoría, ya que ni siquiera conocía la existencia del electrón;  fue dos años después de descubrir el electrón, en 1899, cuando Thomson sostuvo que las partículas emitidas en el efecto fotoeléctrico producido por la luz ultravioleta eran electrones, siendo tal afirmación  aceptada por la comunidad de físicos, asumiendo además, la denominación de fotoelectrones y  aceptando que su generación podía deberse tanto a la luz ultravioleta como a radiación de cualquier otra frecuencia.  En 1902, Philipp Lenard, ayudante de Hertz, experimentó el efecto fotoeléctrico entre dos placas metálicas colocadas en un tubo donde se había hecho el vacío, donde observó cómo al colocar una batería se detectaba corriente eléctrica debido a la emisión de fotoelectrones desde la placa iluminada con luz ultravioleta. Por tanto, antes de 1905 se habían establecido experimentalmente tres hechos empíricos asociados al efecto fotoeléctrico, que son la existencia de una frecuencia umbral de la luz incidente por debajo de la cual no se observa la emisión de fotoelectrones, la energía de los fotoelectrones aumenta con la frecuencia de la luz incidente y la ausencia de tiempo de retardo en la emisión de fotoelectrones. En este mismo año, 1905,  Einstein explicó el efecto fotoeléctrico introduciendo la teoría cuántica de la luz, pero admitió que su teoría también era incompleta porque no podía justificar fenómenos tales como la interferencia, difracción y polarización, mientras que la teoría ondulatoria de Lenard y antecesores sí era capaz de interpretar; debido a esto en 1909, Einstein habló de la fusión de ambas teorías. No fue hasta 1923 que los experimentos de P. Debye y Compton hicieron que la mayoría de los físicos se convencieron en adoptar la teoría cuántica de la luz de Einstein. Ese mismo año Louis Victor Pierre Raymond de Broglie dio un paso más y generalizó la dualidad onda-corpúsculo al caso de las partículas materiales y en concreto de los electrones asignándoles una “onda asociada ficticia”, es decir, en 1923 se generalizó la naturaleza híbrida de los cuantos, dotada al mismo tiempo de características corpusculares y ondulatorias (dualidad onda-corpúsculo). Einstein advirtió de este fenómeno casi 20 años antes y le concedieron, no sin polémica, el Premio Nobel de 1921 por el descubrimiento de la ley del efecto fotoeléctrico.</p>



In [11]:
const mjs = require('./math.js');
mjs.createUnit('gauss', '0.0001 T')
B = mjs.unit(3250, 'gauss')
B.to('T')
q = mjs.elementaryCharge
m = mjs.electronMass
E = mjs.unit(72090E-19, 'J')
E.toJSON()
v = mjs.sqrt(mjs.multiply(2, mjs.divide(E,m)))
x = mjs.divide(mjs.multiply(m, mjs.pow(v, 2)), mjs.multiply(q, v, B))
x.toString()

 

'0.002200914915865607 m'